terraform/ec2/mac/main.tf

```diff
@@ -32,6 +32,40 @@ locals {
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
 }
 
+#####################################################################
+# Prepare Parameters Tests
+#####################################################################
+
+locals {
+  validator_config        = "parameters.yml"
+  final_validator_config  = "final_parameters.yml"
+  cloudwatch_agent_config = "agent_config.json"
+  instance_temp_directory = "/tmp"
+}
+
+resource "local_file" "update-validation-config" {
+  content = replace(file("${var.test_dir}/${local.validator_config}"),
+  "<cloudwatch_agent_config>", "${local.instance_temp_directory}/${local.cloudwatch_agent_config}")
+
+  filename = "${var.test_dir}/${local.final_validator_config}"
+}
+
+// Build and uploading the validator to spending less time in 
+// and avoid memory issue in allocating memory 
+resource "null_resource" "upload-validator" {
+  provisioner "local-exec" {
+    command = <<-EOT
+    cd ../../.. 
+    make validator-build
+    aws s3 cp ./build/validator/darwin/${var.arc}/validator s3://${var.s3_bucket}/integration-test/validator/${var.cwa_github_sha}/darwin/${var.arc}/validator
+    EOT
+  }
+
+  triggers = {
```

Comment:
Always build the validator binaries and [uploading to S3 with trigger](https://ilhicas.com/2019/08/17/Terraform-local-exec-run-always.html)